### ISSUE TITLE
parsed-literal creates bad latex math

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#929]
If I do a parsed literal block like

```
.. parsed-literal::

   abc = :math:`\theta`
```

the result in html is just fine, but the result in latex/pdf is to pull the theta up to the front of and outside of the block.  The generated latex is

```
$\theta$
\begin{Verbatim}[commandchars=\\\{\}]
abc =
\end{Verbatim}
```

when it should be

```
\begin{Verbatim}[commandchars=\\\{\}]
abc = \(\theta\)
\end{Verbatim}
```

Any plans to fix this any time soon?  If not, I'm motivated enough to try to work on this if someone was willing to give me a bit of a sketch about what's causing this and how to fix it.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/929
- Originally reported by: [ross_lippert](https://bitbucket.org/ross_lippert)
- Originally created at: 2012-05-17T17:53:38.788

**Comment by shimizukawa** [comment]
_From [ross_lippert](https://bitbucket.org/ross_lippert) on 2012-05-18 14:04:07+00:00_

I've examined the sources and I can see what is going on.  Then the latex translator visits a literal block it turns on a "string accumulator" called self.verbatim.  As subsequent test blocks are visited, rather than appending to self.body the text is added to self.verbatim.  Once the translator departs the literal block the accumulated self.verbatim text is highlighted and appended to self.body with appropriate latex Verbatim environment delimiters around it.  Of course, this means that any non-text blocks visited between the visit literal and depart literal have already been appended to self.body and are output ahead of the latex Verbatim block, as observed.

I believe that this is a serious misdesign of the latex writer.  I think it can be corrected by abandoning this self.verbatim accumulator strategy and instead doing something more like what the html writer is doing.  You check to see if you are in a parsed-literal by comparing node.rawsource to node.astext() and then, if they are equal, you highlight and dump out the rawsource in a latex Verbatim environment and skip the node, and if they are not equal, you just \begin the Verbatim environment, allow normal processing to continue and then, upon departure, \end the latex Verbatim environment.

Secondly, to make everything work out in latex, the basic math visitors need to use ( and ) instead of $ as their math delimiter and use \sb and \sp instead of _ and ^.

I could carry this out on current sources and submit a patch if this would be useful to you.  Otherwise, I'll look into writing a builder that works the way I want.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-11 09:19:19+00:00_

Closes #929: Support parsed-literal blocks in LaTeX output correctly.

→ <<cset 0103ff2a2b612994b077d9d1e37692ba896d86dc>>

**Comment by shimizukawa** [comment]
_From François Bissey on 2014-05-29 21:37:32+00:00_

Sorry to comment on a closed issue. The use of ( and ) causes problem to us to generate pdf in sage [https://github.com/cschwan/sage-on-gentoo/issues/210#issuecomment-44474361]. Its use break things at least in chapter headings where (...) is apparently fragile. Is there something we are supposed to do (apart from not using math in chapter titles)?
